### PR TITLE
[WIP] Game of Life in C

### PR DESCRIPTION
**Pull request by chrboe**
This is just a preliminary WIP commit which will be squashed and edited
later. I figured I'd open a rather early pull request to allow for more
streamlined code review and to avoid duplicate work.

Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [ ] I named the pull request using `<Sample Program> in <Language>` format
- [ ] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])
- [ ] I fixed #your-issue-number-here

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

**Comment by chrboe** [comment]
@jrg94 not sure how to feel about this, so I'll let you decide.

Basically the problem is that a "square" (as in `n` x `n` fields) field looks really ugly in a terminal. This is because the symbols are about twice as high as they are wide, so it looks really squashed.

My temporary solution to this was to take the width via the command line and divid e it by 2 to get the height. Not really according to the rules, but it looks a lot better imo.

Now that's the decision to be made: do we want actual square fields, or fields that *look* like a square?

**Comment by chrboe** [comment]
@jrg94 I'd say I'm ready for a review at this point. It'd be great if you could take a look and notify me when you're done, so I have a chance to squash those ugly commits away

**Comment by jrg94** [comment]
I'll take a look shortly. Don't worry about squashing the commits. You can do that here during the merge.

File: archive/c/c/game-of-life.c [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+    msec_per_frame = (1./fps) * 1000;
+
+    sleep_time = (struct timespec) {
+        .tv_sec = msec_per_frame / 1000,
+        .tv_nsec = ((long)msec_per_frame % 1000) * 1000000
+    };
+
+    /*
+     * Although the contribution guide specifies that the field should be
```
Comment (jrg94) [inline archive/c/c/game-of-life.c]: I'm glad you made a note of this distinction for the future.

File: archive/c/c/game-of-life.c [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+{
+    uint8_t field_size;
+    uint8_t width, height;
+    int rc;
+    double spawn_rate;
+    uint16_t fps;
+    struct timespec sleep_time;
+    double msec_per_frame;
+
+    if (argc < 4) {
```
Comment (jrg94) [inline archive/c/c/game-of-life.c]: Great stuff here! This helped me get to testing quickly.

**Comment by chrboe** [comment]
Online editors probably won't work, since the output field probably can't handle the ANSI escape sequences.

I suspect this only works locally in a terminal emulator, though I'd love to be proven wrong.

**Comment by jrg94** [comment]
That may be an issue as well, I'm not sure. Personally, I was have problems getting some of the editors to even accept command line arguments. If they did accept command line arguments, they had trouble letting the program run. It seems many of these tools won't let you run a program and output text concurrently. They seem to want the program to execute quickly. Who knows. Probably a memory issue.

**Comment by chrboe** [comment]
Yeah, as I said, the ANSI escape sequences can be a little icky (`cmd` is [known](https://stackoverflow.com/a/16799175) to have issues with it). Unfortunately (depending on your definition of the word), I'm not much of a Windows guy, so somebody else would probably be better off tackling this issue.

**Comment by jrg94** [comment]
We forgot something! 😆 Can we add "Game of Life in C" to the local README?

**Comment by chrboe** [comment]
Oops :smile: Hold on, I'll follow up

**Pull request merged**